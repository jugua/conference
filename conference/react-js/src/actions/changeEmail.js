import axios from 'axios';
import { changeEmailUrl } from '../constants/backend-url';
import { expiredSession,
  emailAlreadyExists,
  technicalError } from '../constants/errors';

const errorHandler = ({ response: { status } }) => {
  switch (status) {
  case 401:
    return expiredSession;
  case 409:
    return emailAlreadyExists;
  default:
    return technicalError;
  }
};

const changeEmail = mail => axios.post(changeEmailUrl, { mail })
  .then(() => ({ message: 'Please, check your email!' }))
  .catch(res => ({ error: errorHandler(res) }));

export default changeEmail;
