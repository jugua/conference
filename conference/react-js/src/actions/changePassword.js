import axios from 'axios';
import { changePasswordUrl } from '../constants/backend-url';
import { wrongPassword,
  expiredSession,
  technicalError } from '../constants/errors';

const errorHandler = ({ response: { status, data: { error } } }) => {
  switch (status) {
  case 401:
    return expiredSession;

  case 400:
    if (error === 'wrong_password') {
      return wrongPassword;
    }
    return technicalError;
  default:
    return technicalError;
  }
};

const changeEmail = data => axios.post(changePasswordUrl, data)
  .then(() => ({ message: 'Your password changed!' }))
  .catch(res => ({ error: errorHandler(res) }));

export default changeEmail;
