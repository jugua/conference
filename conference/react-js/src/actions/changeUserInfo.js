import axios from 'axios';
import { userInfo } from '../constants/backend-url';
import { expiredSession, technicalError } from '../constants/errors';
import actionTypes from '../constants/actions-types';

const editUser = update => ({
  type: actionTypes.SET_USER,
  payload: update,
});

const errorHandler = ({ response: { status } }) => {
  if (status === 401) {
    return expiredSession;
  }
  return technicalError;
};

const success = (dispatch, updatedUser) => {
  dispatch(editUser(updatedUser));
  return { message: 'Your info updated' };
};

const changeUserInfo = dispatch => updatedUser => (
  axios.post(userInfo, updatedUser)
    .then(() => success(dispatch, updatedUser))
    .catch(res => ({ error: errorHandler(res) }))
);

export default changeUserInfo;
