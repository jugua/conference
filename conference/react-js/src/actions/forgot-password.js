import axios from 'axios';

import ACTIONS_TYPES from '../constants/actions-types';
import { forgotPassword } from '../constants/backend-url';

const SUCCESS_MESSAGE = 'SUCCESS_MESSAGE';
const ERROR_MESSAGE = 'ERROR_MESSAGE';

const forgotPasswordAction = (type, payload) => ({ type, payload });

const setForgotPasswordError = (error = '') => forgotPasswordAction(
  ACTIONS_TYPES.SET_FORGOT_PASSWORD_ERROR,
  error,
);

const setForgotPasswordMessage = (message = '') => forgotPasswordAction(
  ACTIONS_TYPES.SET_FORGOT_PASSWORD_MESSAGE,
  message,
);

const setForgotPasswordVisibility =
  (visibility = false) => dispatch => dispatch(forgotPasswordAction(
    ACTIONS_TYPES.SET_FORGOT_PASSWORD_VISIBILITY,
    visibility,
  ));

const sendForgotPasswordEmail = mail => dispatch => (
  axios.post(forgotPassword, { mail })
    .then(() => dispatch(setForgotPasswordMessage(SUCCESS_MESSAGE)))
    .catch(() => dispatch(setForgotPasswordError(ERROR_MESSAGE)))
);

const clearForgotPasswordError = () => dispatch => dispatch(
  setForgotPasswordError(),
);

export default {
  clearForgotPasswordError,
  sendForgotPasswordEmail,
  setForgotPasswordVisibility,
};
