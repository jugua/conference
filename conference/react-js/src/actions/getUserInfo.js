import axios from 'axios';

import actionTypes from '../constants/actions-types';
import { myInfo } from '../constants/backend-url';

const setUser = user => ({
  type: actionTypes.EDIT_USER,
  payload: user,
});

const setUserKeys = user => ({
  type: actionTypes.SET_USER_KEYS,
  payload: Object.keys(user),
});

const success = (dispatch, user) => {
  dispatch(setUser(user));
  dispatch(setUserKeys(user));
};

const getUserInfo = dispatch => (
  axios.get(myInfo)
    .then(res => success(dispatch, res.data))
    .catch(err => err)
);

export default getUserInfo;
