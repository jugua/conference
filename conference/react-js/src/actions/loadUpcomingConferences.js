import axios from 'axios';
import ACTION_TYPES from '../constants/actions-types';

const setUpcomingConferences = conferences => ({
  type: ACTION_TYPES.SET_UPCOMING_CONFERENCES,
  payload: conferences,
});

export const loadUpcomingConferences =
  () => dispatch => axios.get('/conference/upcoming')
    .then(({ data }) => dispatch(setUpcomingConferences(data)));

export const clearUpcomingConferences =
  () => setUpcomingConferences([]);
