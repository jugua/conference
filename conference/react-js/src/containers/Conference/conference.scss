.tabs-title {
	text-align: right;
	font-size: 23px;
	padding-left: 65px;
}

.conference_button {
  background: #39C2D7;
  color: #fff;
  font-size: 18px;
  padding: 3px 15px;
  text-decoration: none;
  position: absolute;
  top: 110px;
  text-transform: uppercase;
}
