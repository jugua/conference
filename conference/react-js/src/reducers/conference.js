import actions from '../constants/actions-types';

const { SET_CONFERENCE } = actions;

const conference = (dataConference = {}, action) => {
  const { type, payload } = action;
  if (type === SET_CONFERENCE) {
    return payload;
  }
  return dataConference;
};

export default conference;
