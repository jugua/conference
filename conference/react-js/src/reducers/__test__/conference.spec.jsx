import conference from '../conference';
import actions from '../../constants/actions-types';

const { SET_CONFERENCE } = actions;

describe('REDUCER onference', () => {
  const initialState = {};
  it('SET_CONFERENCE', () => {
    const action = {
      type: SET_CONFERENCE,
      payload: {
        id: 1,
        title: 'some',
      },
    };
    const state = conference(initialState, action);
    expect(state).toMatchSnapshot();
  });
  it('initialState', () => {
    const state = conference(
      initialState,
      { type: undefined, payload: undefined },
    );
    expect(state).toMatchSnapshot();
  });
});
