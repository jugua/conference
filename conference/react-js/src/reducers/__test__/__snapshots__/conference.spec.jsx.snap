// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`REDUCER onference SET_CONFERENCE 1`] = `
Object {
  "id": 1,
  "title": "some",
}
`;

exports[`REDUCER onference initialState 1`] = `Object {}`;
